libfreerdp/core/mcs.c

```diff
@@ -601,26 +629,48 @@ BOOL mcs_send_connect_initial(rdpMcs* mcs)
 {
 	int status;
 	int length;
-	wStream* s;
+	wStream* s = NULL;
 	int bm, em;
-	wStream* gcc_CCrq;
-	wStream* client_data;
+	wStream* gcc_CCrq = NULL;
+	wStream* client_data = NULL;
 
 	mcs_initialize_client_channels(mcs, mcs->settings);
 
 	client_data = Stream_New(NULL, 512);
+	if (!client_data)
+	{
+		WLog_ERR(TAG, "Stream_New failed!");
+		return FALSE;
+	}
 	gcc_write_client_data_blocks(client_data, mcs);
 
 	gcc_CCrq = Stream_New(NULL, 1024);
+	if (!gcc_CCrq)
+	{
+		status = -1;
```

Comment:
we could initialize status to -1 so that all following error treatments would not have to do a status = -1...